src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -99,13 +152,16 @@ void make_target_dir(mp::SSHSession& session, const std::string& root, const std
 // Assume it is already created.
 void set_owner_for(mp::SSHSession& session, const std::string& root, const std::string& relative_target)
 {
-    auto vm_user = run_cmd(session, "id -nu");
-    auto vm_group = run_cmd(session, "id -ng");
-    mp::utils::trim_end(vm_user);
-    mp::utils::trim_end(vm_group);
+    auto vm_user = run_string_cmd(session, "id -nu");
+    auto vm_group = run_string_cmd(session, "id -ng");
 
-    run_cmd(session, fmt::format("sudo /bin/bash -c 'cd \"{}\" && chown -R {}:{} {}'", root, vm_user, vm_group,
-                                 relative_target.substr(0, relative_target.find_first_of('/'))));
+    // Get the first directory of the relative path.
+    std::string::size_type first_slash = relative_target.find_first_of('/');
+    std::string first_dir =
+        (std::string::npos == first_slash) ? relative_target : relative_target.substr(0, first_slash);
+
+    run_cmd(session,
+            fmt::format("sudo /bin/bash -c 'cd \"{}\" && chown -R {}:{} {}'", root, vm_user, vm_group, first_dir));
```

Comment:
Why this change? FWIW we try to avoid, what we call, "drive-by" changes. If you want to refactor something unrelated to the fix / feature you're working on, please file a separate PR.

Also just realized these should be numeric IDs, not strings, otherwise we need to cater for spaces in them.

Reply:
#1443 deals with the numer IDs issue.

Reply:
It's a good thing to avoid drive-by changes. I'll forget about this change, which is targeted at making the function more robust in case `relative_target` does not have any slashes.